### PR TITLE
Facility LDAP class implementation

### PR DESCRIPTION
**Pull request by mvocu**
Introduce Facility class into LDAPc:
- add related attribute and objectclass definition to LDAP schema
- add facility model class with perun bean attribute descriptions
- add facility bean for model with additional attributes
- add facility synchronizer class
- hook up facility related audit log events

Pre-existing attributes related to facility model that are being stored in resource entries continue to do so, in addition to storing them in facility entries.

### CODE REVIEW

File: perun-ldapc/src/main/resources/perun-ldapc.xml [hunk 1]
```diff
@@ -45,6 +45,14 @@ http://www.springframework.org/schema/aop http://www.springframework.org/schema/
 	<bean id="deletionEventProcessor" class="cz.metacentrum.perun.ldapc.processor.impl.DeletionEventProcessor">
 		<property name="dispatchConditions">
 			<list>
+				<bean class="cz.metacentrum.perun.ldapc.processor.impl.RegexpDispatchEventCondition">
+					<property name="beansCondition">
+						<list>
+							<value>cz.metacentrum.perun.core.api.Facility</value>
+						</list>
+					</property>
+					<property name="pattern" value=" deleted.$" />
```
Comment (zlamalp) [inline perun-ldapc/src/main/resources/perun-ldapc.xml]: We have again wrong message in AuditEvent. Keep the regex here and update produced message``FacilityDeleted`` class in project ``perun-base``, package ``cz.metacentrum.perun.audit.events.FacilityManagerEvents``. Thank you.

File: perun-ldapc/src/main/resources/perun-ldapc.xml [hunk 2]
```diff
@@ -92,6 +100,14 @@ http://www.springframework.org/schema/aop http://www.springframework.org/schema/
 	<bean id="creationEventProcessor" class="cz.metacentrum.perun.ldapc.processor.impl.CreationEventProcessor">
 		<property name="dispatchConditions">
 			<list>
+				<bean class="cz.metacentrum.perun.ldapc.processor.impl.RegexpDispatchEventCondition">
+					<property name="beansCondition">
+						<list>
+							<value>cz.metacentrum.perun.core.api.Facility</value>
+						</list>
+					</property>
+					<property name="pattern" value=" created.$" />
```
Comment (zlamalp) [inline perun-ldapc/src/main/resources/perun-ldapc.xml]: We have again wrong message in AuditEvent. Keep the regex here and update produced message``FacilityCreated`` class in project ``perun-base``, package ``cz.metacentrum.perun.audit.events.FacilityManagerEvents``. Thank you.

File: perun-ldapc/src/main/resources/perun-ldapc.xml [hunk 3]
```diff
@@ -598,6 +622,34 @@ http://www.springframework.org/schema/aop http://www.springframework.org/schema/
 		</property>
 	</bean>
 
+	<bean id="perunFacility" class="cz.metacentrum.perun.ldapc.model.impl.PerunFacilityImpl">
+		<property name="attributeDescriptions">
+			<!--  this list will be added to built-in defaultAttributeDescriptions -->
+			<list>
```
Comment (pajavyskocil) [inline perun-ldapc/src/main/resources/perun-ldapc.xml]: Here is the list of attributes, which we need in LDAP by default :

urn:perun:facility:attribute-def:def:OIDCClientID
urn:perun:facility:attribute-def:def:CheckGroupMembership
urn:perun:facility:attribute-def:virt:voShortNames
urn:perun:facility:attribute-def:def:wayfFilter
urn:perun:facility:attribute-def:def:wayfEFilter
urn:perun:facility:attribute-def:def:entityID
urn:perun:facility:attribute-def:def:isTestSp
urn:perun:facility:attribute-def:def:proxyIdentifiers
urn:perun:facility:attribute-def:def:showOnServiceList

Comment (zlamalp) [reply perun-ldapc/src/main/resources/perun-ldapc.xml]: Just a note - for ``facility:virt:voShortNames`` we need to update attribute module in order to generate messages about changes.

Comment (mvocu) [reply perun-ldapc/src/main/resources/perun-ldapc.xml]: Please write the requirements to the respective Redmine issues (there are two, AFAIK - one specifically for facility class use-cases, one requesting the implementation).

File: perun-ldapc/src/main/resources/perun-ldapc.xml (hunk 2 above)
Comment (mvocu) [reply perun-ldapc/src/main/resources/perun-ldapc.xml]: Updated

File: perun-ldapc/src/main/resources/perun-ldapc.xml (hunk 1 above)
Comment (mvocu) [reply perun-ldapc/src/main/resources/perun-ldapc.xml]: Updated

**Pull request merged**